cpu/native/socket_zep/socket_zep.c

```diff
@@ -431,30 +433,9 @@ void socket_zep_setup(socket_zep_t *dev, const socket_zep_params_t *params)
         _send_zep_hello(dev);
     }
 
-    /* generate hardware address from local address */
-    uint8_t ss_array[sizeof(struct sockaddr_storage)] = { 0 };
-    socklen_t ss_len = sizeof(struct sockaddr_storage);
```

Comment:
I would prefer keeping the old generation as a fallback (or at least a variant of it). CPU-ID in native is based on the PID, so it would change with every restart.

Reply:
Huh, but the old generation is not stable either.
On master:

```
main(): This is RIOT! (Version: 2021.01-devel-1650-g66466)
RIOT network stack example application
All up, running the shell now
> ifconfig
ifconfig
Iface  7  HWaddr: CA:5C  Channel: 26  NID: 0x23 
          Long HWaddr: 00:5A:45:50:0A:00:CA:5C 
          L2-PDU:102  MTU:1280  HL:64  RTR  
          6LO  IPHC  
          Source address length: 8
          Link type: wireless
          inet6 addr: fe80::25a:4550:a00:ca5c  scope: link  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff00:ca5c
          
          Statistics for Layer 2
            RX packets 0  bytes 0
            TX packets 2 (Multicast: 2)  bytes 43
            TX succeeded 2 errors 0
          Statistics for IPv6
            RX packets 0  bytes 0
            TX packets 2 (Multicast: 2)  bytes 128
            TX succeeded 2 errors 0

> reboot
reboot


		!! REBOOT !!

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2021.01-devel-1650-g66466)
RIOT network stack example application
All up, running the shell now
> ifconfig
ifconfig
Iface  7  HWaddr: E3:B9  Channel: 26  NID: 0x23 
          Long HWaddr: 00:5A:45:50:0A:00:E3:B9 
          L2-PDU:102  MTU:1280  HL:64  RTR  
          6LO  IPHC  
          Source address length: 8
          Link type: wireless
          inet6 addr: fe80::25a:4550:a00:e3b9  scope: link  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff00:e3b9
          
          Statistics for Layer 2
            RX packets 0  bytes 0
            TX packets 2 (Multicast: 2)  bytes 43
            TX succeeded 2 errors 0
          Statistics for IPv6
            RX packets 0  bytes 0
            TX packets 2 (Multicast: 2)  bytes 128
            TX succeeded 2 errors 0
```

(see #15589)

Reply:
Ah right, because you meddled with the local endpoint. I remember! But I don't think the PID is suitable for EUI-64 generation as a fall-back either (as it is too short).

Reply:
> But I don't think the PID is suitable for EUI-64 generation as a fall-back either (as it is too short).

But luid will take care of extending it to the size of an EUI-64 - and by it's very nature, it is guaranted to be unique on a system.

Reply:
Ok, I just checked. There seems to be enough entropy in `luid` so that the EUI-64 of two different `native` instances are very different. More even than with the old approach. Then let's go for the standard fallback :-).